tools/shared/pragma.c

```diff
@@ -1184,37 +1184,27 @@ do_sw(void)
       return true;
     break;
   case SW_UNROLL:
-    /* [no]unroll		-x/y 11 3
-     * [no]unroll = c	-x/y 11 1
-     * [no]unroll = n	-x/y 11 2
-     *     unroll = c:v     -y   11 3  -x    9 v
-     *     unroll = n:v     -y   11 3  -x   10 v
+    /* unroll       -x 11 0x1   -y 11 0x402
+     * unroll = n   -x 11 0x2   -y 11 0x401 -x 9 n
+     * unroll (n)   -x 11 0x2   -y 11 0x401 -x 9 n
+     * nounroll     -x 11 0x400 -y 11 0x3
      */
     typ = gtok();
-    if (typ != T_EQUAL) {
-      if (no_specified)
+    // !dir$ [no]unroll
+    if (typ == T_END) {
+      if (no_specified) { // !dir$ nounroll
         bset(DIR_OFFSET(currdir, x[11]), 0x400);
-      else
-        bset(DIR_OFFSET(currdir, x[11]), 0x3);
-    } else if (gtok() == T_IDENT) {
-      if (strcmp(ctok, "c") == 0)
-        i = 9;
-      else if (strcmp(ctok, "n") == 0)
-        i = 10;
-      else
-        return true;
-      if (no_specified)
-        bset(DIR_OFFSET(currdir, x[11]), i - 8);
-      else if (gtok() != T_COLON)
-        bclr(DIR_OFFSET(currdir, x[11]), i - 8);
-      else if (gtok() != T_INT)
-        return true;
-      else {
-        if (itok <= 0)
-          itok = 1;
-        assn(DIR_OFFSET(currdir, x[i]), (int)itok);
-        bclr(DIR_OFFSET(currdir, x[11]), 3);
+        bclr(DIR_OFFSET(currdir, x[11]), 0x3);
+      } else { // !dir$ unroll
+        bset(DIR_OFFSET(currdir, x[11]), 0x1);
+        bclr(DIR_OFFSET(currdir, x[11]), 0x402);
       }
+    } else if (typ == T_EQUAL || typ == T_LP) {
```

Comment:
for the T_LP case should we check for T_RP as well to ensure well formedness?

Reply:
Some existing code in a different place does the following. Would it make sense to do something similar?

```
    if (typ == T_LP) {
      if (gtok() != T_INT)
        return true;
      i = itok;
      if (gtok() != T_RP)
        return true;
      typ = gtok();
    } 
```

Reply:
Yeah, we have written something like that internally (i.e. failing silently), but we are trying to see if we could make it better and produce a syntax error like what clang would do if the `T_RP` is missing, e.g.
```
clang-unroll-pragma.c:4:36: error: expected ')'
  #pragma clang loop unroll_count(8
                                   ^
1 error generated.
```
But in the interest of time, we could just do the simpler thing and get this PR merged, and plan to improve the error reporting in the future.

Reply:
I am fine with all the improvements coming in a new patch. Approving this one.